extensions/amp-video/0.1/amp-video.js

```diff
@@ -765,7 +765,18 @@ export class AmpVideo extends AMP.BaseElement {
    */
   setUpCaptions_() {
     const captionsId = this.element.getAttribute('captions-id');
-    if (!captionsId) {
+    // if (!captionsId) {
+    //   return;
+    // }
+
+    const trackEl = this.element.querySelector('track');
+    if (!captionsId && trackEl) {
+      // Works because this function is called twice, so this function runs again after it is attached from the first time it's called.
+      const captionsId = trackEl.getAttribute('src');
+      const captionsEl = this.win.document.createElement('amp-story-captions');
```

Comment:
Nit: If you use preact it should be much simpler, `<amp-story-captions id={captionsId}></amp-story-captions>`